lib/nfc/protocols/mf_classic/mf_classic_poller.c

```diff
@@ -829,6 +1033,1081 @@ NfcCommand mf_classic_poller_handler_key_reuse_read_sector(MfClassicPoller* inst
+                FURI_LOG_I(
+                    TAG,
+                    "Found key candidate %06llx",
+                    bit_lib_bytes_to_num_be(key_candidate->data, sizeof(MfClassicKey)));
+                dict_attack_ctx->current_key = *key_candidate;
+                dict_attack_ctx->reuse_key_sector = (target_block / 4);
+                dict_attack_ctx->current_key_type = target_key_type;
+                free(key_candidate);
+                break;
+            } else {
+                free(dict_attack_ctx->nested_nonce.nonces);
+                dict_attack_ctx->nested_nonce.nonces = NULL;
+                dict_attack_ctx->nested_nonce.count = 0;
+            }
+        }
+
+        FURI_LOG_D(
+            TAG,
+            "Target: %u (key type %s, block %u) cuid: %08lx",
+            dict_attack_ctx->nested_target_key,
+            (target_key_type == MfClassicKeyTypeA) ? "A" : "B",
+            target_block,
+            cuid);
+    } while(false);
+
+    instance->state = MfClassicPollerStateNestedController;
+
+    mf_classic_poller_halt(instance);
+    return command;
+}
+
+NfcCommand mf_classic_poller_handler_nested_log(MfClassicPoller* instance) {
+    furi_assert(instance->mode_ctx.dict_attack_ctx.nested_nonce.count > 0);
+    furi_assert(instance->mode_ctx.dict_attack_ctx.nested_nonce.nonces);
+
+    NfcCommand command = NfcCommandContinue;
+    bool params_saved = false;
+    MfClassicPollerDictAttackContext* dict_attack_ctx = &instance->mode_ctx.dict_attack_ctx;
+    Storage* storage = furi_record_open(RECORD_STORAGE);
+    Stream* stream = buffered_file_stream_alloc(storage);
+    FuriString* temp_str = furi_string_alloc();
+    bool weak_prng = dict_attack_ctx->prng_type == MfClassicPrngTypeWeak;
+    bool static_encrypted = dict_attack_ctx->static_encrypted;
+
+    do {
+        if(weak_prng && (!(static_encrypted)) && (dict_attack_ctx->nested_nonce.count != 2)) {
+            FURI_LOG_E(
+                TAG,
+                "MfClassicPollerStateNestedLog expected 2 nonces, received %u",
+                dict_attack_ctx->nested_nonce.count);
+            break;
+        }
+
+        uint32_t nonce_pair_count = dict_attack_ctx->prng_type == MfClassicPrngTypeWeak ?
+                                        1 :
+                                        dict_attack_ctx->nested_nonce.count;
+
+        if(!buffered_file_stream_open(
+               stream, MF_CLASSIC_NESTED_LOGS_FILE_PATH, FSAM_WRITE, FSOM_OPEN_APPEND))
+            break;
+
+        bool params_write_success = true;
+        for(size_t i = 0; i < nonce_pair_count; i++) {
+            MfClassicNestedNonce* nonce = &dict_attack_ctx->nested_nonce.nonces[i];
+            // TODO: Avoid repeating logic here
+            uint8_t nonce_sector = nonce->key_idx / (weak_prng ? 4 : 2);
+            MfClassicKeyType nonce_key_type =
+                (nonce->key_idx % (weak_prng ? 4 : 2) < (weak_prng ? 2 : 1)) ? MfClassicKeyTypeA :
```

Comment:
Maybe this should be replaced? According to comments

Reply:
@RebornedBrain I was asking if I could submit those as a separate PR to further refactor the code and improve performance (to be merged in 1.2), as I don't want to miss the merge window for 1.1 to give users a working process. To the best of my knowledge it's stable.

Reply:
@noproto ok, no problem